Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.cpp

```diff
@@ -1126,11 +1126,24 @@ int CLuaEngineDefs::EngineGetModelTextures(lua_State* luaVM)
     if (argStream.HasErrors())
         return luaL_error(luaVM, argStream.GetFullErrorMessage());
 
+    CLuaMain*  pLuaMain = m_pLuaManager->GetVirtualMachine(luaVM);
+    CResource* pParentResource = pLuaMain ? pLuaMain->GetResource() : NULL;
+
+    if (!pParentResource)
+    {
+        lua_pushnil(luaVM);
+        return 1;
+    }
```

Comment:
Is `pParentResource` ever actually nil?

Reply:
I don't think so. This check is not needed. @theSarrum Please remove this check, we can merge this PR for 1.5.8.